nixos/modules/services/networking/dhcpcd.nix

```
@@ -207,13 +207,19 @@ in
 
         serviceConfig =
           { Type = "forking";
-            PIDFile = "/run/dhcpcd.pid";
+            PIDFile = "/run/dhcpcd/pid";
```

Is this the upstream default? I would expect it to be like this

```suggestion
            PIDFile = "/run/dhcpcd/dhcpcd.pid";
```

---

yes :/ though technically the upstream default is `/var/run/dhpcd/pid`, this elides that `/var/run -> /run` link

---

Does dhcpd manage this directory?

It seems we should also explicitly manage this with RuntimeDirectory, that would prevent situations where dhcpd might forget to clean up after itself.

---

it does, but setting RuntimeDirectory does make sense anyway.